libraries/reconstruction_cuda/cuda_gpu_multidim_array.cu

```diff
@@ -102,4 +102,189 @@ T interpolatedElementBSpline2D_Degree3(T x, T y, int xdim, int ydim, T* data)
 
 #undef LOOKUP_TABLE_LEN
 
+#define LOOKUP_TABLE_LEN 6
+
+/*
+* This version doesn't subtract shift from position in float and doesn't lose
+* precision with higher values of `x` and `y`
+*/
+
+template<typename T>
+__device__
+T interpolatedElementBSpline2D_Degree3New(int x, int y, T x_shift, T y_shift, int xdim, int ydim, T* data)
```

Comment:
Is anybody using the original implementation? 
What is the difference between the original one and this one?